### ISSUE TITLE
consistent-function-scoping: regression: ignored in function argument

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#770]
The `consistent-function-scoping` rule no longer warns about functions defined inside a function being passed as an argument.  For example:

```js
process.nextTick(() => {
  function returnsZero() {
    return true;
  }
  process.exitCode = returnsZero();
});
```

In `eslint-plugin-unicorn@19.0.1` this would cause:

    example.js:2:3: Move function `returnsZero` to the outer scope. [Error/unicorn/consistent-function-scoping]

In `eslint-plugin-unicorn@20.0.0` and later (up to 7781c3f, current `master`) it does not cause an error.

I bisected the regression to 1f4413d from #707.  Perhaps the IIFE identification is a bit too aggressive?  (I wouldn't consider any of the functions in the above example to be immediately-invoked, but perhaps I'm missing something?)

Thanks again,
Kevin

**Comment by fisker** [comment]
Sorry about this, It's mistake in #707, should be fixed by #772

**Comment by kevinoid** [comment]
No worries.  I should have caught it earlier.  I just tested #772 and it fixes this issue for me and I didn't find any other regressions.  Thanks @fisker!
